diff --git a/apps/Makefile b/apps/Makefile
index 02b705ddb2..558caa5f49 100644
--- a/apps/Makefile
+++ b/apps/Makefile
@@ -49,6 +49,8 @@ include $(ASTTOPDIR)/Makefile.moddir_rules
 
 $(call MOD_ADD_C,app_confbridge,$(wildcard confbridge/*.c))
 
+$(call MOD_ADD_C,app_rpt,$(wildcard app_rpt/rpt_*.c))
+
 app_confbridge.o: _ASTCFLAGS+=$(AST_NO_FORMAT_TRUNCATION)
 app_meetme.o: _ASTCFLAGS+=$(AST_NO_FORMAT_TRUNCATION)
 app_minivm.o: _ASTCFLAGS+=$(AST_NO_FORMAT_TRUNCATION)
diff --git a/channels/Makefile b/channels/Makefile
index 0af82d9ff0..9337710728 100644
--- a/channels/Makefile
+++ b/channels/Makefile
@@ -30,6 +30,10 @@ $(call MOD_ADD_C,chan_sip,$(wildcard sip/*.c))
 $(call MOD_ADD_C,chan_pjsip,$(wildcard pjsip/*.c))
 $(call MOD_ADD_C,chan_dahdi,$(wildcard dahdi/*.c) sig_analog.c sig_pri.c sig_ss7.c)
 
+chan_simpleusb.so: LIBS+=-lusb -lasound
+chan_usbradio.so: LIBS+=-lusb -lasound
+chan_beagle.so: LIBS+=-lasound
+
 chan_dahdi.o: _ASTCFLAGS+=$(call get_menuselect_cflags,LOTS_OF_SPANS)
 chan_mgcp.o: _ASTCFLAGS+=$(AST_NO_FORMAT_TRUNCATION)
 chan_unistim.o: _ASTCFLAGS+=$(AST_NO_FORMAT_TRUNCATION)
